### PR TITLE
Check Yubikey OTP length before validating

### LINKED ISSUE #3746: Check Yubikey OTP length before validating
According to the docs the Yubikey OTP is always 44 chars long (https://developers.yubico.com/OTP/OTPs_Explained.html).
We can check the length of the given OTP before  doing any further validation (this is also true for HOTP/TOTP/Email/SMS token).
That way we can identify possible broken input before getting cryptic errors like "CRC checksum failed".

### PR DESCRIPTION
**Pull request by jona-samuel**
closes #3746

### CODE REVIEW

**Comment by jona-samuel** [comment]
should we implement this for totp, hotp, SMS and email too?

File: privacyidea/lib/decorators.py [hunk 1]
```diff
@@ -46,6 +46,26 @@ def token_locked_wrapper(*args, **kwds):
     return token_locked_wrapper
 
 
+def check_token_otp_lenght(func):
```
Comment (plettich) [inline privacyidea/lib/decorators.py:49]: please correct to `check_token_otp_length()`

File: privacyidea/lib/tokens/emailtoken.py [hunk 2]
```diff
@@ -328,6 +328,7 @@ def create_challenge(self, transactionid=None, options=None):
 
     @log_with(log)
     @check_token_locked
+    @check_token_otp_lenght
```
Comment (plettich) [inline privacyidea/lib/tokens/emailtoken.py:331]: please move this decorator one line up. Decorators are applied from bottom to top (innermost first). We should first check if the token is locked and then if the otp length matches.

File: privacyidea/lib/tokens/hotptoken.py [hunk 3]
```diff
@@ -400,6 +400,7 @@ def _calc_otp(self, counter):
 
     @log_with(log)
     @check_token_locked
+    @check_token_otp_lenght
```
Comment (plettich) [inline privacyidea/lib/tokens/hotptoken.py:403]: see above

File: privacyidea/lib/tokens/smstoken.py [hunk 4]
```diff
@@ -368,6 +368,7 @@ def create_challenge(self, transactionid=None, options=None):
 
     @log_with(log)
     @check_token_locked
+    @check_token_otp_lenght
```
Comment (plettich) [inline privacyidea/lib/tokens/smstoken.py:371]: and all others...

File: privacyidea/lib/tokens/yubikeytoken.py [hunk 5]
```diff
@@ -280,6 +283,7 @@ def check_otp(self, anOtpVal, counter=None, window=None, options=None):
             otp_bin = modhex_decode(yubi_otp)
         except KeyError:
             # The OTP value is no yubikey aes otp value and can not be decoded
+            log.debug("OTP could not be decode")
```
Comment (plettich) [inline privacyidea/lib/tokens/yubikeytoken.py:286]: please change to `could not be decoded`

**Review by plettich — CHANGES_REQUESTED** [review]

**Review by plettich — APPROVED** [review]

**Pull request merged**